src/JustEat.StatsD/StatsDMessageFormatter.cs

```diff
@@ -104,34 +104,46 @@ public string Increment(long magnitude, double sampleRate, params string[] statB
         public string Gauge(double magnitude, string statBucket)
         {
             var stat = string.Format(_cultureInfo, "{0}{1}:{2}|g", _prefix, statBucket, magnitude);
-            return Format(stat, DefaultSampleRate);
+            return Format(DefaultSampleRate, stat);
         }
         public string Gauge(double magnitude, string statBucket, DateTime timestamp)
         {
             var stat = string.Format(_cultureInfo, "{0}{1}:{2}|g|@{3}", _prefix, statBucket, magnitude, timestamp.AsUnixTime());
-            return Format(stat, DefaultSampleRate);
+            return Format(DefaultSampleRate, stat);
         }
 
         public string Gauge(long magnitude, string statBucket)
         {
             var stat = string.Format(_cultureInfo, "{0}{1}:{2}|g", _prefix, statBucket, magnitude);
-            return Format(stat, DefaultSampleRate);
+            return Format(DefaultSampleRate, stat);
         }
 
         public string Gauge(long magnitude, string statBucket, DateTime timestamp)
         {
             var stat = string.Format(_cultureInfo, "{0}{1}:{2}|g|@{3}", _prefix, statBucket, magnitude, timestamp.AsUnixTime());
-            return Format(stat, DefaultSampleRate);
+            return Format(DefaultSampleRate, stat);
         }
 
         public string Event(string name)
         {
             return Increment(name);
         }
 
-        private string Format(String stat, double sampleRate)
+        private string Format(double sampleRate, string stat)
         {
-            return Format(sampleRate, stat);
+            if (sampleRate < DefaultSampleRate)
+            {
+                if (Random.NextDouble() <= sampleRate)
```

Comment:
I’m not following what the `Random` is for?

Reply:
It is to give us a random number between 0 and 1 so that we can filter out a percentage of them, thus sampling. We also put the sample rate into the payload we send cause StatsD allow us to indicate that there has been down-sampling at the source.

Reply:
The functionality about random and sample rate should not have changed with this PR